### PR TITLE
Added announcement editing

### PR DESCRIPTION
**Pull request by DmytroZelinsky**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by nicoolle — COMMENTED** [review]
Remove code smell, please

**Review by nicoolle — APPROVED** [review]

File: EPlast/EPlast.BLL/Services/GoverningBodies/Announcement/GoverningBodyAnnouncementService.cs [hunk 1]
```diff
@@ -87,5 +87,19 @@ public async Task<List<string>> GetAllUserAsync()
 
             return userIds;
         }
+        public async Task<int> EditAnnouncement(GoverningBodyAnnouncementUserDTO annoncement)
```
Comment (OleksandrBilan) [inline EPlast/EPlast.BLL/Services/GoverningBodies/Announcement/GoverningBodyAnnouncementService.cs:90]: Typo in word "annoncement" as input parameter

File: EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs [hunk 2]
```diff
@@ -601,6 +601,36 @@ public async Task GetById_ReturnNoContent()
             Assert.IsNotNull(result);
             Assert.IsInstanceOf<NotFoundResult>(result);
         }
+        [Test]
+        public async Task EditAnnouncement_ModeStatIsValid_ReturnsOk()
```
Comment (OleksandrBilan) [inline EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs:605]: I think it should be "ModelStateIsValid"

File: EPlast/EPlast.Tests/Services/GoverningBody/Announcement/GoverningBodyAnnouncementServiceTest.cs [hunk 3]
```diff
@@ -202,6 +202,26 @@ public async Task GetAllUserAsync_Valid()
             Assert.IsInstanceOf<List<string>>(result);
         }
 
+        [Test]
+        public async Task EditAnnouncement_ReturnsId()
+        {
+            //Arrange
+            _repoWrapper
+                .Setup(x => x.GoverningBodyAnnouncement.Update(It.IsAny<GoverningBodyAnnouncement>()))
+                .Callback(() => { });
```
Comment (OleksandrBilan) [inline EPlast/EPlast.Tests/Services/GoverningBody/Announcement/GoverningBodyAnnouncementServiceTest.cs:211]: You don't need to write this Callback

**Review by VolodymyrKostiv — APPROVED** [review]

File: EPlast/EPlast.Tests/Services/GoverningBody/Announcement/GoverningBodyAnnouncementServiceTest.cs (hunk 3 above)
Comment (DmytroZelinsky) [reply EPlast/EPlast.Tests/Services/GoverningBody/Announcement/GoverningBodyAnnouncementServiceTest.cs:211]: very need!

Comment (OleksandrBilan) [reply EPlast/EPlast.Tests/Services/GoverningBody/Announcement/GoverningBodyAnnouncementServiceTest.cs:211]: 💩

File: EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs [hunk 4]
```diff
@@ -601,6 +601,36 @@ public async Task GetById_ReturnNoContent()
             Assert.IsNotNull(result);
             Assert.IsInstanceOf<NotFoundResult>(result);
         }
+        [Test]
+        public async Task EditAnnouncement_ModeStateIsValid_ReturnsOk()
```
Comment (petromonka) [inline EPlast/EPlast.Tests/Controllers/GoverningBodiesControllerTests.cs:605]: It's a typo? Shouldn't there be "Mode**l**State"?

**Review by petromonka — APPROVED** [review]

**Pull request merged**